py3/pyxtension/streams.py

```diff
@@ -371,6 +371,12 @@ def __unique_generator(itr, f: Optional[Callable[[_K], _V]]):
                 if m_el not in st:
                     st.add(m_el)
                     yield el
+    
+    def on_next(self, f: Callable[[_K], None]) -> 'stream[_V]':
```

Comment:
In general I avoid using inner functions where it's possible to replace them with instance methods or static functions, since they add an extra layer of computational resource complexity as they are generated at the runtime and consume some additional memory. Being a library code, which is used in lots of projects, computational performance is an important aspect of `stremarate` library.